src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/BlockSyntaxExtensions.cs

```diff
@@ -62,6 +62,10 @@ internal static class BlockSyntaxExtensions
                 block.TryConvertToExpressionBody(languageVersion, preference, out var expression, out semicolonToken))
             {
                 arrowExpression = SyntaxFactory.ArrowExpressionClause(expression);
+                arrowExpression = arrowExpression.WithPrependedLeadingTrivia(
+                        block.GetAncestors().FirstOrDefault().GetLeadingTrivia());
```

Comment:
I think we should be more specific in which ancestors we're looking for, particularly the get / set  accessors.  Not sure if there is a common syntax node that we could look for instead of explicitly looking for the get/set (cc @CyrusNajmabadi )

Reply:
talked offline - we can check if the parent is an accessor which should cover the get/set/init kinds (I believe the actual node kind is `AccessorDeclarationSyntax` but we'll have to check the tree).

We should definitely also add test cases that exercise it in `set` and `init` scenarios too.

Reply:
It looks like the 'use expression body' code fix is only offered when there's a get accessor, it doesn't show for set or init accessors. Should I make the check for the three scenarios despite this?

Reply:
Ah of course, my mistake.  We can only convert the entire thing to an expression body if it only has a get.  Lets just check for the get accessor then.